push_test.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	smartling "github.com/Smartling/api-sdk-go"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+type request struct {
+	response string
+	code     int
+}
+
+type roundTripFunc func(req *http.Request) *http.Response
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+func mockHttpClient(fn roundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: fn,
+	}
+}
+
+func TestPushStopUnauthorized(t *testing.T) {
```

Comment:
Is it common for go have 1 letter params name? I'm ok for with a short name for counters or params in lambdas, but the current one is function.